docs/snippets/react/page-story-args-within-story.js.mdx

```diff
@@ -0,0 +1,32 @@
+```js
+// my-component/component.stories.js|jsx
+
+import { useArgs } from '@storybook/preview-api';
+import { Switch } from '.';
+
+export default {
+  title: 'Inputs/Switch',
+  component: Switch,
+};
+
+export const Example = {
+  args: {
+    isChecked: false,
+    label: 'Switch Me!',
+  },
+  /**
+   * 👇 React believes the Storybook function "useArgs" is a hook, but it's really not.
+   * Using a named capital-letter function prevents an ESLint warning related to this.
+   * Don't lint? You can use an arrow function instead.
+   */
```

Comment:
@jonthenerd are you ok with simplifying this to the following and applying it to the TS versions (e.g., pre-4.9 and 4.9:

```js
/**
 * 👇 To avoid linting issues, it is recommended to use a function with a capitalized name or an arrow function
 */
 ```

Reply:
@jonniebigodes I simplified it, but didn't keep it exactly as you provided as it read as if using an arrow function would pass linting (which it would not). I also added the typescript 4.9 version. This should be ready for re-review.